llvm/lib/Analysis/DomConditionCache.cpp

```diff
@@ -34,23 +34,39 @@ static void findAffectedValues(Value *Cond,
     }
   };
 
-  ICmpInst::Predicate Pred;
-  Value *A;
-  if (match(Cond, m_ICmp(Pred, m_Value(A), m_Constant()))) {
-    AddAffected(A);
+  bool TopLevelIsAnd = match(Cond, m_LogicalAnd());
```

Comment:
Is there a reason it doesn't also work for bitwise and of conditions?

Reply:
See also the tests `test_cond_and` and `test_cond_logical_and` that check both bitwise and logical operations.